app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -2,15 +2,18 @@
 
 # rubocop:disable Metrics/ClassLength
 class GroupAssignmentInvitationsController < ApplicationController
+  class InvalidStatusForRouteError < StandardError; end
+
   include InvitationsControllerMethods
 
   layout "layouts/invitations"
 
-  before_action :check_group_not_previous_acceptee,      only: [:show]
-  before_action :check_user_not_group_member,            only: [:show]
-  before_action :check_should_redirect_to_roster_page,   only: [:show]
-  before_action :authorize_group_access,                 only: [:accept_invitation]
-  before_action :ensure_github_repo_exists,              only: [:successful_invitation]
+  before_action :route_based_on_status,                  only: %i[setupv2 successful_invitation]
+  before_action :check_group_not_previous_acceptee,      only: :show
+  before_action :check_user_not_group_member,            only: :show
+  before_action :check_should_redirect_to_roster_page,   only: :show
+  before_action :authorize_group_access,                 only: :accept_invitation
+  before_action :ensure_github_repo_exists,              only: :successful_invitation
```

Comment:
Only real change here is L11
```diff
+ before_action :route_based_on_status,                  only: %i[setupv2 successful_invitation]
```
The rest of the changes in this diff are removing square brackets when redundant.